fdbserver/workloads/SpecialKeySpaceCorrectness.actor.cpp

```diff
@@ -300,6 +300,105 @@ struct SpecialKeySpaceCorrectnessWorkload : TestWorkload {
+				throw;
+			ASSERT(e.code() == error_code_illegal_tenant_access);
+			tenantTx->reset();
+		}
+		// tenant clear
+		try {
+			tenantTx->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
+			tenantTx->clear(SpecialKeySpace::getManagementApiCommandRange("exclude"));
+			ASSERT(false);
+		} catch (Error& e) {
+			if (e.code() == error_code_actor_cancelled)
+				throw;
+			ASSERT(e.code() == error_code_illegal_tenant_access);
+			tenantTx->reset();
+		}
+		// tenant check that conflict ranges stay the same after commit
+		// and depending on if RYW is disabled
+		try {
+			state RangeResult readresult1;
+			state RangeResult readresult2;
+			try {
+				tenantTx->setOption(FDBTransactionOptions::READ_YOUR_WRITES_DISABLE);
+				tenantTx->addReadConflictRange(singleKeyRange(LiteralStringRef("testKeylll")));
+				tenantTx->addWriteConflictRange(singleKeyRange(LiteralStringRef("testKeylll")));
+				wait(store(readresult1, tenantTx->getRange(readConflictRangeKeysRange, CLIENT_KNOBS->TOO_MANY)));
+				wait(tenantTx->commit());
+				CODE_PROBE(true, "conflict range tenant commit succeeded");
+			} catch (Error& e) {
+				if (e.code() == error_code_actor_cancelled)
+					throw;
+				CODE_PROBE(true, "conflict range tenant commit error thrown");
+			}
+			wait(store(readresult2, tenantTx->getRange(readConflictRangeKeysRange, CLIENT_KNOBS->TOO_MANY)));
+			ASSERT(readresult1 == readresult2);
+			tenantTx->reset();
+		} catch (Error& e) {
+			throw;
+		}
+		// proper conflict ranges
+		loop {
+			try {
+				tenantTx->setOption(FDBTransactionOptions::READ_YOUR_WRITES_DISABLE);
+				tenantTx2->setOption(FDBTransactionOptions::READ_YOUR_WRITES_DISABLE);
+				wait(success(tenantTx->getReadVersion()));
+				wait(success(tenantTx2->getReadVersion()));
+				tenantTx->addReadConflictRange(singleKeyRange(LiteralStringRef("foo")));
+				tenantTx->addWriteConflictRange(singleKeyRange(LiteralStringRef("foo")));
+				tenantTx2->addWriteConflictRange(singleKeyRange(LiteralStringRef("foo")));
+				wait(tenantTx2->commit());
+				try {
+					wait(tenantTx->commit());
+					ASSERT(false);
+				} catch (Error& e) {
+					if (e.code() != error_code_not_committed) {
+						wait(tenantTx->onError(e));
+						continue;
+					}
+					// Read conflict ranges of tenantTx and check for "foo" with no tenant prefix
+					state RangeResult readConflictRange =
+					    wait(tenantTx->getRange(readConflictRangeKeysRange, CLIENT_KNOBS->TOO_MANY));
+					state RangeResult writeConflictRange =
+					    wait(tenantTx->getRange(writeConflictRangeKeysRange, CLIENT_KNOBS->TOO_MANY));
+
+					std::string readExpected = "foo"_sr.withPrefix(readConflictRangeKeysRange.begin).toString();
+					std::string writeExpected = "foo"_sr.withPrefix(writeConflictRangeKeysRange.begin).toString();
+					ASSERT(readConflictRange.begin()->key.toString().find(readExpected) != std::string::npos);
+					ASSERT(writeConflictRange.begin()->key.toString().find(writeExpected) != std::string::npos);
```

Comment:
In both of these cases, there should only be the single key in the conflict ranges. You could check that the first item is exactly that key and that there is nothing else in the returned result.

Additionally, we should check the third module that reports the conflicting keys from the transaction and verify that this key is also in that set and doesn't have the tenant prefix.